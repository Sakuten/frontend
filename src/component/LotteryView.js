import React from 'react'
import { observer } from 'mobx-react'

const LotteryView = ({list, classroom, onChange}) => (
  <div>
    {
      list.length !== 0
        ? list.filter(c => c.classroom_id === Number(classroom))
          .map(performN => <PerformanceNumber onChange={onChange} performN={performN} key={performN} />)
        : <span data-test='lottery-notfound'>現在応募は受け付けておりません。</span>
    }
  </div>
)

export class PerformanceNumber extends React.Component {
  componentWillMount () {
    this.props.onChange(this.props.performN.id)
    console.log(this.props)
  }
  render () {
    const { performN } = this.props
    return <span data-test='lottery-lottery' key={performN.id} value={performN.id}>第{performN.index + 2}公演</span>
  }
}
export default observer(LotteryView)
