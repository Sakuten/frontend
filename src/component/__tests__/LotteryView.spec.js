import React from 'react'
import { shallow } from 'enzyme'
import LotteryView, { PerformanceNumber } from '../LotteryView'

const setup = propOverrides => {
  const props = Object.assign({
    list: [
      {
        'classroom_id': 1,
        'done': false,
        'id': 4,
        'index': 0,
        'name': '5A.0'
      },
      {
        'classroom_id': 2,
        'done': false,
        'id': 5,
        'index': 0,
        'name': '5B.0'
      }
    ],
    classroom: 1,
    onChange: jest.fn()
  }, propOverrides)

  const wrapper = shallow(<LotteryView {...props} />)

  return {
    props,
    wrapper,
    lottery: wrapper.find(PerformanceNumber),
    notfound: wrapper.find('[data-test="lottery-notfound"]'),
  }
}

describe('components', () => {
  describe('LotteryView', () => {
    it('render', () => {
      const { wrapper } = setup()
      expect(wrapper).toMatchSnapshot()
    })

    it('renders one lottery', () => {
      const { lottery } = setup()
      expect(lottery.length).toBe(1)
      expect(lottery.props().performN.id).toBe(4)
    })

    it('renders with human-readable messages', () => {
      const { lottery } = setup()
      expect(lottery.html()).toBe('<span data-test=\"lottery-lottery\" value=\"4\">第2公演</span>')
    })

    it('renders only in same classroom', () => {
      const { lottery } = setup({classroom: 2})
      expect(lottery.length).toBe(1)
      expect(lottery.props().performN.id).toBe(5)
    })

    it('renders a message when the list was empty', () => {
      const { notfound } = setup({list: []})
      expect(notfound.length).toBe(1)
    })
  })
})
