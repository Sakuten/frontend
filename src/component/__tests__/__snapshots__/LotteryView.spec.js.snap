// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components LotteryView render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LotteryView
    classroom={1}
    list={
      Array [
        Object {
          "classroom_id": 1,
          "done": false,
          "id": 4,
          "index": 0,
          "name": "5A.0",
        },
        Object {
          "classroom_id": 2,
          "done": false,
          "id": 5,
          "index": 0,
          "name": "5B.0",
        },
      ]
    }
    onChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PerformanceNumber
          onChange={[MockFunction]}
          performN={
            Object {
              "classroom_id": 1,
              "done": false,
              "id": 4,
              "index": 0,
              "name": "5A.0",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "[object Object]",
        "nodeType": "class",
        "props": Object {
          "onChange": [MockFunction],
          "performN": Object {
            "classroom_id": 1,
            "done": false,
            "id": 4,
            "index": 0,
            "name": "5A.0",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PerformanceNumber
            onChange={[MockFunction]}
            performN={
              Object {
                "classroom_id": 1,
                "done": false,
                "id": 4,
                "index": 0,
                "name": "5A.0",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "[object Object]",
          "nodeType": "class",
          "props": Object {
            "onChange": [MockFunction],
            "performN": Object {
              "classroom_id": 1,
              "done": false,
              "id": 4,
              "index": 0,
              "name": "5A.0",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
